azurerm/internal/services/web/resource_arm_function_app.go

```diff
@@ -870,3 +945,24 @@ func flattenFunctionAppSiteCredential(input *web.UserProperties) []interface{} {
 
 	return append(results, result)
 }
+
+func flattenFunctionAppIpRestriction(input *[]web.IPSecurityRestriction) []interface{} {
+	restrictions := make([]interface{}, 0)
+
+	if input == nil {
+		return restrictions
+	}
+
+	for _, v := range *input {
+		block := make(map[string]interface{})
+		if ip := v.IPAddress; ip != nil {
+			block["ip_address"] = *ip
+		}
+		if vNetSubnetID := v.VnetSubnetResourceID; vNetSubnetID != nil {
+			block["subnet_id"] = *vNetSubnetID
```

Comment:
we need to ensure there's a value set for this nested field even if it's unset - as such can we make this:

```suggestion
			subnetId := ""
			if v.VnetSubnetResourceID != nil {
				subnetId = *v.VnetSubnetResourceID
			}
			block["subnet_id"] = subnetID
```